lib/plugins/threaded-signing/plugin.c

```diff
@@ -254,30 +283,32 @@ threaded_openssl_get_signature(sv_threaded_plugin_t *self,
   SignedVideoReturnCode status = SV_OK;
 
   g_mutex_lock(&self->mutex);
-  if (self->plugin_state == THREADED_SIGNING_HAS_SIGNATURE && self->signature_info) {
-    if (self->signature_info->signature_size > max_signature_size) {
+  if (self->output_buffer[0].signing_error) {
```

Comment:
Won't this crash for you? Element 0 is a NULL pointer until the first signature is ready.
You need to check if there are any signatures in the buffer first.